cherry_picker/cherry_picker/cherry_picker.py

```diff
@@ -72,10 +73,16 @@ def upstream(self):
 
     @property
     def sorted_branches(self):
+        def version_from_branch(branch):
+            try:
+                return tuple(map(int, re.match(r'^.*(?P<version>\d+(\.\d+)+).*$', branch).groupdict()['version'].split('.')))
+            except AttributeError as attr_err:
+                raise ValueError(f'Branch {branch} seems to not have a version in its name.') from attr_err
+
         return sorted(
```

Comment:
@Mariatta optionally we could try/except this as well and fall back to returning branches list in their original order.

Reply:
@Mariatta what do you think about falling back to unordered branch list?

Reply:
Since there is no need to support unordered branch list yet, I'm thinking to keep things as is and not change the behavior.